src/ferui-components/tree-view/interfaces.ts

```diff
@@ -0,0 +1,87 @@
+import { TemplateRef } from '@angular/core';
+
+/**
+ * Tree Node Data Interface
+ */
+export interface TreeNodeData<T> {
+  data: T;
+  field: string;
+}
+
+/**
+ * Paging Params Interface
+ */
+export interface PagingParams {
+  offset: number;
+  limit: number;
+}
+
+/**
+ * Tree Node Data Retriever Interface for client-side tree view
+ */
+export interface TreeNodeDataRetriever<T> {
+  // Get promise that returns TreeNodeData children array from Tree Node
+  getChildNodeData(parent: TreeNodeData<T>): Promise<Array<TreeNodeData<T>>>;
+  // Get promise that returns whether TreeNode has children nodes
+  hasChildNodes(parent: TreeNodeData<T>): Promise<boolean>;
+  // If the developer wants to give us an icon template for the expansion/collapse of node
+  getIconTemplate?(): TemplateRef<any>;
+  // If developer wishes to give us a template ref we shall render this template in the view
+  getNodeTemplate?(): TemplateRef<any>;
+}
+
+/**
+ * Paged Tree Node Data Retriever for server-side tree view
+ */
+export interface PagedTreeNodeDataRetriever<T> extends TreeNodeDataRetriever<T> {
+  getPagedChildNodeData(parent: TreeNodeData<T>, pagingParams: PagingParams): Promise<Array<TreeNodeData<T>>>;
+}
+
+/**
+ * Non root tree node empty singleton instance
+ */
+export class NonRootTreeNode implements TreeNodeData<any> {
+  public static INSTANCE: NonRootTreeNode = new NonRootTreeNode();
```

Comment:
you need a private constructor

Reply:
Agree with cory and please, use lowercased variable within a class, even if it's static.